modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/TrafficSources.js

```diff
@@ -41,27 +43,64 @@ const FALLBACK_COLOR = '#e92563';
 const getColorByName = (name) => COLORS_MAP[name] || FALLBACK_COLOR;
 
 export default function TrafficSources({
+	dataProvider,
 	languageTag,
 	onTrafficSourceClick,
-	trafficSources,
 }) {
 	const [highlighted, setHighlighted] = useState(null);
 
-	const [, addWarning] = useWarning();
+	const [warning, addWarning] = useWarning();
+
+	const {validAnalyticsConnection} = useContext(ConnectionContext);
 
 	const [{publishedToday}] = useContext(StoreContext);
 
-	const fullPieChart = trafficSources.some((source) => !!source.value);
+	const [trafficSources, setTrafficSources] = useState([]);
+
+	const isMounted = useIsMounted();
+
+	useEffect(() => {
+		if (validAnalyticsConnection) {
+			dataProvider()
+				.then((trafficSources) => {
+					if (isMounted()) {
+						setTrafficSources(trafficSources);
+					}
+				})
+				.catch(() => {
+					setTrafficSources([]);
+					if (isMounted()) {
+						if (!warning) {
+							addWarning();
+						}
+					}
+				});
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [dataProvider, validAnalyticsConnection, warning]);
+
+	const fullPieChart = useMemo(
+		() => trafficSources.some((source) => !!source.value),
+
+		[trafficSources]
+	);
 
-	const missingTrafficSourceValue = trafficSources.some(
-		(trafficSource) => trafficSource.value === undefined
+	const missingTrafficSourceValue = useMemo(
+		() =>
+			trafficSources.some(
+				(trafficSource) => trafficSource.value === undefined
+			),
+		[trafficSources]
 	);
 
 	useEffect(() => {
 		if (missingTrafficSourceValue) {
-			addWarning();
+			if (!warning) {
+				addWarning();
+			}
 		}
-	}, [addWarning, missingTrafficSourceValue]);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Again, why this? If it's complaining about `addWarning` that one should have a stable identity because it comes from a `useCallback` and `dispatch` (the input to that `useCallback` dependency array) will be stable too. Which is to say, it looks safe to add and drop the lint suppression.

Reply:
👍